Benchmarks_on_OVS_integration_approaches.md

```diff
@@ -0,0 +1,213 @@
+# Benchmarks on OVS integration approaches
+
+OpenStack used to use the legacy Nova-network to provide a basic flat network model. However in its successor Neutron, this model is extended as a flat network cannot meet the requirements of a modern data center. Neutron uses a single bridge called the “integration bridge” that all guest instances are connected to provide connectivity to an external network with a physical network interface. This is essentially the same as the nova-network used to connect VMs to a physical interface.
+
+In QEMU, a L2 network interface is simulated with a tap device. However, in XenServer, vif is more frequently used because of its performance is much better. As Neutron heavily depends on tap device, it seems to be a conflict for the integration between XenServer and OpenStack. [XenAPI: OVS agent updates the wrong port with Neutron](https://review.openstack.org/#/c/242846/) introduces a new bridge called "interim bridge" to work between the guest vm and the integration bridge so vif will still be used while without invasively changing Neutron. This blog post is aiming to investigate the performance cost of "interim bridge" by setting up an environment comparing the conditions with and without interim bridge.
```

Comment:
-simulated +emulated
Possibly better to explain what actually happens:  When we boot an HVM guest, we start with both a tap and a vif device attached to the VM.  The emulated tap device is used by the VM until the paravirtualised tools have been loaded, at which point the more performant VIF device is used and the tap device is deleted.

Neutron does not heavily depend on a "tap device" in this context; Neutron works with a single connection to the VM and does not cope with multiple devices beign used for the same connection